### ISSUE TITLE
Dependency provider fails to provide default falsy value

### ISSUE THREAD

**Issue by approxit** [ets-labs/python-dependency-injector#550]
I found that `providers.Dependency` is not recognizing falsy default value as any provided value, which failbacks to raising errors due to missing dependency definition.

Simplest test case that works without errors:
``` python
default = {'foo': 'bar'}
provider = providers.Dependency(instance_of=dict, default=default)
self.assertEqual(provider(), default)
```
But on falsy default value it complaints about no dependency defined:
``` python
default = {}
provider = providers.Dependency(instance_of=dict, default=default)
self.assertEqual(provider(), default)  # Raises dependency_injector.errors.Error: Dependency is not defined
```
I'm expecting that any provided default value will be accepted and used when no explicit value is passed to `providers.Dependency`. I'm using `dependency_injector` version of `4.37.0`.

**Comment by rmk135** [comment]
Hey @approxit ,

Thanks for reporting the issue. It looks like a bug, and I agree that the second example should not raise an exception. I'll get it fixed

**Comment by rmk135** [comment]
Fixed in dev version

**Comment by rmk135** [comment]
Fixed in version ``4.38.0``
